extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java

```diff
@@ -278,6 +230,97 @@ private void consumeService(Class<?> serviceClass, BiConsumer<Class<?>, Collecti
         }
     }
 
+    private void resolveLiquibaseResources(
+            CurateOutcomeBuildItem curateOutcome,
+            BuildProducer<ServiceProviderBuildItem> services,
+            BuildProducer<ReflectiveClassBuildItem> reflective,
+            BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitialized,
+            BuildProducer<NativeImageResourceBuildItem> resource) {
+
+        var dependencies = curateOutcome.getApplicationModel().getDependencies();
+        var liquibaseDependency = dependencies.stream()
+                .filter(this::isLiquibaseDependency)
+                .findFirst()
+                .orElseThrow(() -> new IllegalStateException("Liquibase dependency not found"));
+
+        var tree = liquibaseDependency.getContentTree();
+        loadLiquibaseRootProperties(tree, resource);
+        loadLiquibaseXsdResources(tree, resource);
+        loadLiquibaseServiceProviderConfig(tree, services, reflective, runtimeInitialized);
+    }
+
+    private boolean isLiquibaseDependency(ResolvedDependency dependency) {
+        return dependency.getResolvedPaths().stream()
+                .map(Path::getFileName)
+                .map(Path::toString)
+                .anyMatch(s -> s.contains(LIQUIBASE_ARTIFACT_NAME));
```

Comment:
Wouldn't `dependency.getArtifactId().equals(LIQUIBASE_ARTIFACT_NAME)` work here?

Reply:
You're right, somehow I managed to overlook that possibility.
I will change this accordingly.

Reply:
Please do that and also test the groupId. Typically you could have someone having a `my-groupid:liquibase-core` in their app and you don't want to catch that.

Also I recommend using the `CONSTANT.equals(variable)` pattern instead of `variable.equals(CONSTANT)`. In this case, the variables won't be null so it won't change anything but it's a good practice to avoid NPEs. Once you have this habit, you're a lot safer.